runtime/src/message_processor.rs

```diff
@@ -302,22 +303,28 @@ impl<'a> ThisInvokeContext<'a> {
     }
 }
 impl<'a> InvokeContext for ThisInvokeContext<'a> {
-    fn push(&mut self, key: &Pubkey) -> Result<(), InstructionError> {
-        if self.program_ids.len() > self.bpf_compute_budget.max_invoke_depth {
+    fn push(
+        &mut self,
+        keyed_accounts: &[KeyedAccount],
+        key: &Pubkey,
+    ) -> Result<&[KeyedAccount], InstructionError> {
```

Comment:
Instead of returning the caller's keyed_accounts here it would be more flexible to add a `get_callers_keyed_accounts` accessor